transformer_engine/pytorch/transformer.py

```diff
@@ -95,12 +95,12 @@ def __init__(
 
         self.norm_factor = norm_factor
         self.attention_dropout_ctx = attention_dropout_ctx
+        self.layer_number = layer_number
 
         self.scale_mask_softmax = FusedScaleMaskSoftmax(
             attn_mask_type,
             attention_mask_func,
-            attention_softmax_in_fp32,
-            layer_number if apply_query_key_layer_scaling else None,
+            layer_number,
```

Comment:
Maybe you need to move that scale to the forward call of the softmax so that you know what type its input is going to be.

Reply:
And then pass `None` if it is not fp16.